File: src/ContableUxser.Domain/Entities/Producto.cs
using ContableUxser.Domain.Common;

namespace ContableUxser.Domain.Entities;

public class Producto : BaseEntity
{
    public Guid EmpresaId { get; set; }
    public string CodigoBarras { get; set; } = string.Empty;
    public string Nombre { get; set; } = string.Empty;
    public decimal CostoPromedio { get; set; }
    public decimal PrecioVenta { get; set; }
    public decimal StockActual { get; set; }
    public decimal StockMinimo { get; set; }
    public string? Ubicacion { get; set; }
    public string? ImagenUrl { get; set; }

    public Empresa Empresa { get; set; } = null!;
    public ICollection<VentaDetalle> VentaDetalles { get; set; } = new List<VentaDetalle>();
    public ICollection<MovimientoInventario> MovimientosInventario { get; set; } = new List<MovimientoInventario>();
}

File: contable_uxser_app/lib/domain/entities/producto.dart
class Producto {
  final String id;
  final String empresaId;
  final String codigoBarras;
  final String nombre;
  final double costoPromedio;
  final double precioVenta;
  final double stockActual;
  final double stockMinimo;
  final String? ubicacion;
  final String? imagenUrl;

  Producto({
    required this.id,
    required this.empresaId,
    required this.codigoBarras,
    required this.nombre,
    required this.costoPromedio,
    required this.precioVenta,
    required this.stockActual,
    required this.stockMinimo,
    this.ubicacion,
    this.imagenUrl,
  });

  bool get stockBajo => stockActual <= stockMinimo;
}
